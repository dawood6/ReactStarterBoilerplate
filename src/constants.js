export const INITIAL = {};

export const SET_DATA = 'SET_DATA';
