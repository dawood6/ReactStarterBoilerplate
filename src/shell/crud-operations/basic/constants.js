export const INITIAL = {
  searchData: '',
  selected: [],

  dataset: [
    { id: 1, primary: 'Vanilla', accent: 'MobX' },
    { id: 2, primary: 'Angular', accent: 'NGXS' },
    { id: 3, primary: 'React', accent: 'Redux' },
    { id: 4, primary: 'Vue', accent: 'Vuex' },
  ],

  addData: {},
  editData: {},
  viewData: {},
  deleteData: {},

  dialogs: {
    add: false,
    edit: false,
    view: false,
    delete: false,
  },
};

export const ADD_ITEM = '[crudOperations/basic] ADD_ITEM';
export const EDIT_ITEM = '[crudOperations/basic] EDIT_ITEM';
export const DELETE_ITEM = '[crudOperations/basic] DELETE_ITEM';

export const SET_DATA = '[crudOperations/basic] SET_DATA';
