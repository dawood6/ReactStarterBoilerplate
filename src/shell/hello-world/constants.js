export const INITIAL = {
  hello: 'Hello',
};

export const SET_DATA = '[helloWorld] SET_DATA';
