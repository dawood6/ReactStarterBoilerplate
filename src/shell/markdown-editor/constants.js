export const INITIAL = {
  articleList: [],
  articleDetail: {
    subject: '',
    content: '',
  },
};

export const SET_DATA = '[markdownEditor] SET_DATA';
