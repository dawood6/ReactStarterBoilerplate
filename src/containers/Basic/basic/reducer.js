import { handleActions } from 'redux-actions';

import { INITIAL, ADD_ITEM, EDIT_ITEM, DELETE_ITEM, SET_DATA } from './constants';

export default handleActions({
  [ADD_ITEM](state, { payload: { primary, accent } }) {
    const id = state.dataset.reduce((maxId, item) => Math.max(item.id, maxId), -1) + 1;

    return {
      ...state,
      dataset: [...state.dataset, { id, primary, accent }],
    };
  },
  [EDIT_ITEM](state, { payload: { id, primary, accent } }) {
    return {
      ...state,
      dataset: [
        ...state.dataset
          .map(item => (item.id === id ? { ...item, primary, accent } : item)),
      ],
    };
  },
  [DELETE_ITEM](state, { id }) {
    const removeById = arr => (
      [...arr.filter(item => item.id !== id)]
    );

    return {
      ...state,
      dataset: removeById(state.dataset),
      selected: state.selected.length ? removeById(state.selected) : state.selected,
    };
  },
  [SET_DATA](state, { data }) {
    return { ...state, ...data };
  },
}, INITIAL);
