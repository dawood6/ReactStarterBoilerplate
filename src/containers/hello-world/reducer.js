import { handleActions } from 'redux-actions';

import assignDeep from '~/shared/assign-deep';

import { INITIAL, SET_DATA } from './constants';

export default handleActions({
  [SET_DATA](state, { data }) {
    return { ...assignDeep(state, data) };
  },
}, INITIAL);
