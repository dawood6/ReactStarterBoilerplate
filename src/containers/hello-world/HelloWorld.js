import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { dynamic } from 'redux-dynamic-manager';
import styled from '@emotion/styled';
import { Link } from 'react-router-dom'

import reducer from './reducer';

const Title = styled('h1')`
  color: #222;
`;

const HelloWorld = () => {
  const [world] = useState('World');
  const { hello } = useSelector(state => state.helloWorld);

  return (
    <div id="hello-world">
      <Title>{hello}, {world}!</Title>
      <Link to="/crud-operations/basic">Goto Component</Link>
    </div>
  );
};

export default dynamic('helloWorld', reducer)(HelloWorld);
