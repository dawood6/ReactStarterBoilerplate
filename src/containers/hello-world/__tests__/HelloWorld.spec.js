import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import HelloWorld from '../HelloWorld';

import { INITIAL } from '../constants';
import reducer from '../reducer';

describe('HelloWorld', () => {
  it('should render an initial component', () => {
    const store = createStore(reducer, { helloWorld: { ...INITIAL } });
    const { container } = render(<Provider store={store}><HelloWorld /></Provider>);
    expect(container).toMatchSnapshot();
  });
});
