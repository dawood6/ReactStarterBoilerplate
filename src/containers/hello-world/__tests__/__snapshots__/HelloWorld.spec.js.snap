// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld should render an initial component 1`] = `
.emotion-0 {
  color: #222;
}

<div>
  <div
    id="hello-world"
  >
    <h1
      class="emotion-0 emotion-1"
    >
      Hello
      , 
      World
      !
    </h1>
  </div>
</div>
`;
